// convex/dist/esm/cli/lib/mcp/tools/status.js
"use strict";
import {
  fetchDeploymentCredentialsProvisionProd
} from "../../api.js";
import { deploymentSelectionFromOptions } from "../../api.js";
import { z } from "zod";
import { deploymentDashboardUrlPage } from "../../../dashboard.js";
import { encodeDeploymentSelector } from "../deploymentSelector.js";
const projectDirDescription = `
The root directory of the Convex project. This is usually the editor's workspace directory
and often includes the 'package.json' file and the 'convex/' folder.

Pass this option unless explicitly instructed not to.
`;
const inputSchema = z.object({
  projectDir: z.string().optional().describe(projectDirDescription)
});
const outputSchema = z.object({
  availableDeployments: z.array(
    z.object({
      kind: z.string(),
      deploymentSelector: z.string(),
      url: z.string(),
      dashboardUrl: z.string().optional()
    })
  )
});
const description = `
Get all available deployments for a given Convex project directory.

Use this tool to find the deployment selector, URL, and dashboard URL for each
deployment associated with the project. Pass the deployment selector to other
tools to target a specific deployment.

When deployed to Convex Cloud, projects have a development ({"kind": "ownDev"}) and
production ({"kind": "ownProd"}) deployment. Generally default to using the development
deployment unless you'd specifically like to debug issues in production.

When running locally, there will be a single "urlWithAdminKey" deployment.
`.trim();
export const StatusTool = {
  name: "status",
  description,
  inputSchema,
  outputSchema,
  handler: async (ctx, input) => {
    const projectDir = input.projectDir ?? ctx.options.projectDir;
    if (projectDir === void 0) {
      return await ctx.crash({
        exitCode: 1,
        errorType: "fatal",
        printedMessage: "No project directory provided. Either provide the `projectDir` argument or configure the MCP server with the `--project-dir` flag."
      });
    }
    process.chdir(projectDir);
    const deployment = await deploymentSelectionFromOptions(ctx, ctx.options);
    const credentials = await fetchDeploymentCredentialsProvisionProd(
      ctx,
      deployment
    );
    const availableDeployments = [
      {
        kind: deployment.kind,
        deploymentSelector: encodeDeploymentSelector(projectDir, deployment),
        url: credentials.url,
        dashboardUrl: credentials.deploymentName && deploymentDashboardUrlPage(credentials.deploymentName, "")
      }
    ];
    if (deployment.kind === "ownDev") {
      const prodDeployment = { kind: "ownProd" };
      const prodCredentials = await fetchDeploymentCredentialsProvisionProd(
        ctx,
        prodDeployment
      );
      if (prodCredentials.deploymentName && prodCredentials.deploymentType) {
        availableDeployments.push({
          kind: prodDeployment.kind,
          deploymentSelector: encodeDeploymentSelector(
            projectDir,
            prodDeployment
          ),
          url: prodCredentials.url,
          dashboardUrl: deploymentDashboardUrlPage(
            prodCredentials.deploymentName,
            ""
          )
        });
      }
    }
    return { availableDeployments };
  }
};
